metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -125,6 +125,58 @@ void CanonicalDBG::map_to_nodes(std::string_view sequence,
     }
 }
 
+void CanonicalDBG
+::get_kmers_from_suffix(node_index node, std::vector<node_index> &children) const {
+    const auto &alphabet = graph_.alphabet();
+    node_index next;
+    std::string rev_seq = get_node_sequence(node).substr(1) + std::string(1, '\0');
+    ::reverse_complement(rev_seq.begin(), rev_seq.end());
+    assert(rev_seq[0] == '\0');
+
+    const auto *dbg_succ = dynamic_cast<const DBGSuccinct*>(&graph_);
+    if (dbg_succ) {
+        const auto &boss = dbg_succ->get_boss();
+        dbg_succ->call_nodes_with_suffix_matching_longest_prefix(
+            std::string_view(&rev_seq[1], get_k() - 1),
+            [&](node_index next, uint64_t /* match length */) {
+                auto edge = dbg_succ->kmer_to_boss_index(next);
+                char c = boss.decode(boss.get_minus_k_value(edge, get_k() - 2).first);
+                if (c == boss::BOSS::kSentinel)
+                    return;
+
+                ::reverse_complement(&c, &c + 1);
+                size_t i = boss.encode(c);
+
+                if (children[i] == DeBruijnGraph::npos) {
+                    if (!primary_)
+                        rev_comp_cache_.Put(next, next + offset_);
```

Comment:
Maybe add a comment why you put these to the cache and how you expect it to make things faster.
Like, you expect these nodes to be called soon after this function, or something?

If there is no reason, it's better not to change the cache.

Reply:
There's already a cache for storing the children and parents of each given node, so this is redundant